src/utils/iframe.js

```diff
@@ -0,0 +1,75 @@
+/**
+ * Copyright 2019 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {Deferred} from '../../vendor/ampproject/amphtml/src/utils/promise';
+
+/** @private {?boolean} */
+let srcdocSupported = null;
+
+/**
+ * Returns `true` if srcdoc is supported. https://caniuse.com/#search=srcdoc
+ * @return {boolean}
+ */
+export function isSrcdocSupported() {
+  if (srcdocSupported === null) {
+    srcdocSupported = 'srcdoc' in HTMLIFrameElement.prototype;
+  }
+  return srcdocSupported;
+}
+
+/**
+ * Writes content to given iframe using document.{open, write, close}
+ * @param {!HTMLIFrameElement} iframe
+ * @param {string} content
+ * @return {!Promise}
+ */
+export function writeToIframe(iframe, content) {
+  iframe.src = 'about:blank';
+  const childDoc = iframe.contentWindow.document;
+  childDoc.open();
+  childDoc.write(content);
+  childDoc.close();
+  return whenIframeLoaded(iframe);
+}
+
+/**
+ * Writes content to given iframe using srcdoc attribute.
+ * @param {!HTMLIFrameElement} iframe
+ * @param {string} content
+ * @return {!Promise}
+ */
+export function writeToSrcdoc(iframe, content) {
+  iframe.srcdoc = content;
+  return whenIframeLoaded(iframe);
+}
+
+/**
+ * Returns a promise that resolves on the next `load`
+ * event from the given iframe.
+ * @param {!HTMLIFrameElement} iframe
+ * @return {!Promise}
+ */
+export function whenIframeLoaded(iframe) {
+  const {promise, resolve} = new Deferred();
+  iframe.addEventListener(
+    'load',
+    () => {
+      resolve(iframe);
+    },
+    {once: true}
```

Comment:
😮 TIL

does this cleanup as well?

Reply:
Yep :)

Reply:
![wonderful](https://user-images.githubusercontent.com/254946/60556603-05393080-9cf7-11e9-894b-23b29ed3f5f8.gif)